File: Count of common characters in two strings.java
import java.util.Scanner;

class Main {
    public static void main(String[] a) {
        Scanner s = new Scanner(System.in);
        String x = s.nextLine(), y = s.nextLine();
        int c = 0;
        for (char z : x.toCharArray()) if (y.contains(String.valueOf(z))) { c++; y = y.replaceFirst(String.valueOf(z), ""); }
        System.out.println(c);
    }
}

Example 1:
Input:
china
india
Output:
3
Example 2:
Input:
energy
every
Output:
4    

File: Count of common characters in two strings.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    string x, y;
    getline(cin, x);
    getline(cin, y);
    int c = 0;
    for (char z : x) {
        size_t pos = y.find(z);
        if (pos != string::npos) {
            c++;
            y.erase(pos, 1);
        }
    }
    cout << c;
    return 0;
}

Example 1:
Input:
china
india
Output:
3
Example 2:
Input:
energy
every
Output:
4 
